libparsec/crates/types/src/shamir.rs

```diff
@@ -0,0 +1,168 @@
+use serde::{Deserialize, Serialize};
+use serde_bytes::Bytes;
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(into = "ShamirRecoverySecretData", from = "ShamirRecoverySecretData")]
+pub struct ShamirRecoverySecret {
+    pub data_key: SecretKey,
+    pub reveal_token: InvitationToken,
+}
+
+parsec_data!("schema/shamir/shamir_recovery_secret.json5");
+
+impl_transparent_data_format_conversion!(
+    ShamirRecoverySecret,
+    ShamirRecoverySecretData,
+    data_key,
+    reveal_token,
+);
+
+impl ShamirRecoverySecret {
+    pub fn dump(&self) -> DataResult<Vec<u8>> {
+        ::rmp_serde::to_vec_named(self).map_err(|_| DataError::BadSerialization {
+            format: None,
+            step: "dump shamir recovery secret",
+        })
+    }
+
+    pub fn load(buf: &[u8]) -> DataResult<Self> {
+        ::rmp_serde::from_slice(buf).map_err(|_| DataError::BadSerialization {
+            format: None,
+            step: "load shamir recovery secret",
+        })
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(
+    into = "ShamirRecoveryShareDataData",
+    from = "ShamirRecoveryShareDataData"
+)]
+pub struct ShamirRecoveryShareData {
+    pub weighted_share: Vec<ShamirShare>,
+}
+
+parsec_data!("schema/shamir/shamir_recovery_share_data.json5");
+
+impl_transparent_data_format_conversion!(
+    ShamirRecoveryShareData,
+    ShamirRecoveryShareDataData,
+    weighted_share,
+);
+
+impl ShamirRecoveryShareData {
+    pub fn decrypt_verify_and_load_for(
+        ciphered: &[u8],
+        recipient_privkey: &PrivateKey,
+        author_verify_key: &VerifyKey,
+    ) -> DataResult<ShamirRecoveryShareData> {
+        let signed = recipient_privkey.decrypt_from_self(ciphered)?;
+        let compressed = author_verify_key.verify(&signed)?;
+        let mut serialized = vec![];
+        ZlibDecoder::new(compressed)
+            .read_to_end(&mut serialized)
+            .map_err(|_| DataError::BadSerialization {
+                format: None,
+                step: "compression shamir recovery share data",
+            })?;
+        let data: ShamirRecoveryShareData =
+            rmp_serde::from_slice(&serialized).map_err(|_| DataError::BadSerialization {
+                format: None,
+                step: "shamir recovery share data",
+            })?;
+        Ok(data)
+    }
+
+    pub fn dump_sign_and_encrypt_for(
+        &self,
+        author_signkey: &SigningKey,
+        recipient_pubkey: &PublicKey,
+    ) -> Vec<u8> {
+        let serialized = rmp_serde::to_vec_named(&self).unwrap_or_else(|_| unreachable!());
+        let mut e = ZlibEncoder::new(Vec::new(), flate2::Compression::default());
+        e.write_all(&serialized).unwrap_or_else(|_| unreachable!());
+        let compressed = e.finish().unwrap_or_else(|_| unreachable!());
+        let signed = author_signkey.sign(&compressed);
+        recipient_pubkey.encrypt_for_self(&signed)
+    }
```

Comment:
As mentioned in an earlier comment I would remove the signing from those methods